### ISSUE TITLE
LeakyLogic.AND silently saturates at high p due to safe_pow eps clamping

### ISSUE THREAD

**Issue by HappyHackingOrange** [property-driven-ml/property-driven-ml#9]
## Summary

`LeakyLogic.AND` (the positive-`p` branch of `p_sum`) stops converging to the hard max as `p` grows, and instead saturates at `eps^(1/p)` where `eps = torch.finfo(float32).eps ≈ 1.19e-7`. This is a silent numerical limitation, not an error: `AND` returns a perfectly finite-looking value, but it depends only on `p` and not on the inputs.

Discovered while writing tests for `LeakyLogic`; happy to open a PR with the pinning test below and a fix.

## Root cause

`utils.safe_pow` clamps its base to `finfo.eps`:

```python
def safe_pow(x, y):
    return torch.pow(safe_zero(x), y)   # safe_zero = clamp(x, min=eps)
```

`p_sum` wraps the *outer* power in `safe_pow` too:

```python
def p_sum(self, *xs, p):
    x = torch.stack(xs, dim=0)
    return safe_pow(torch.sum(safe_pow(x, p), dim=0), 1.0 / p)
```

Once `max(x)^p < eps` — i.e. roughly `p · log(1/max(x)) > log(1/eps) ≈ 15.94` in float32 — the inner sum drops below the eps clamp, so the outer call returns `safe_pow(eps, 1/p) = eps^(1/p)` for every element. For `p=50` that's `~0.727`, regardless of input.

## Reproducer

```python
import torch
from property_driven_ml.logics import LeakyLogic

x = torch.tensor([0.3, 0.2])
y = torch.tensor([0.1, 0.15])

# Hard max is [0.3, 0.2]; expect AND to be slightly above that.
print(LeakyLogic(p=50).AND(x, y))
# -> tensor([0.7270, 0.7270])   <- identical, and unrelated to inputs
```

Triggers in the high-`p` regime that is otherwise the whole point of `LeakyLogic` — that's where `(sum x^p)^(1/p)` approaches `max(x)` and the constraint loss is sharpest.

## Expected behavior

For `p=50, x=[0.3, 0.2], y=[0.1, 0.15]`, `AND(x, y)` should be just above the elementwise `max`:
- element 0: `~0.3` (true value `(0.3^50 + 0.1^50)^(1/50) ≈ 0.300`)
- element 1: `~0.2` (true value `(0.2^50 + 0.15^50)^(1/50) ≈ 0.200`)

## Suggested fix

Rewrite `p_sum` in log-domain — the same trick QLL already uses for its LogSumExp AND/OR:

```python
def p_sum(self, *xs, p):
    x = torch.stack(xs, dim=0)
    return torch.exp(torch.logsumexp(p * torch.log(safe_zero(x)), dim=0) / p)
```

That keeps the entire computation numerically stable for any `p > 0` and any positive input.

## Pinning test

Until a fix lands, the saturation can be pinned as known behavior so a future fix is visible:

```python
def test_leaky_and_saturates_at_high_p_due_to_safe_pow():
    x = torch.tensor([0.3, 0.2])
    y = torch.tensor([0.1, 0.15])
    eps = torch.finfo(torch.float32).eps
    expected = eps ** (1.0 / 50.0)
    assert torch.allclose(LeakyLogic(p=50).AND(x, y), torch.full_like(x, expected), atol=1e-5)
```

Reproduced on `main` at version `0.2.7`, Python 3.12, PyTorch 2.5.1+cu121.
